### PR TITLE
feat(event-search): Add projects in heatmaps.

### PR DESCRIPTION
**Pull request by lauryndbrown**

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events.py [hunk 1]
```diff
@@ -196,12 +204,64 @@ def get(self, request, organization):
...
         try:
-            tag_key = tagstore.get_group_tag_keys_and_top_values(
+            tag_keys = tagstore.get_group_tag_keys_and_top_values(
                 project_ids, None, environment_ids, keys=lookup_keys, get_excluded_tags=True, **snuba_args)
         except tagstore.TagKeyNotFound:
             raise ResourceDoesNotExist
 
-        return Response(serialize(tag_key, request.user))
+        if non_tag_lookup_keys:
+            tag_keys.update(self.handle_non_tags_keys(non_tag_lookup_keys, snuba_args))
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_events.py]: should this method be `non_tag_keys`? The double plural is hard to say. :bike:

File: src/sentry/api/endpoints/organization_events.py [hunk 2]
```diff
@@ -196,12 +204,64 @@ def get(self, request, organization):
...
+            if key == 'project':
+                projects = Project.objects.filter(id__in=snuba_args['filter_keys']['project_id'])
+                for project_data in data:
+                    project = projects.filter(id=project_data['project_id'])[0]
+                    project_data['key'] = 'project'
+                    project_data['value'] = project.slug
+            result.add(self.__create_tag_key_tag_value_objects('project', data))
+        return result
+
+    def __query_non_tag_data(self, key, snuba_args):
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_events.py]: What does two `_` do?

File: src/sentry/api/endpoints/organization_events.py [hunk 3]
```diff
@@ -196,12 +204,64 @@ def get(self, request, organization):
...
+        data = raw_query(
+            groupby=[key],
+            aggregations=snuba_args.get('aggregations', []) + [
+                ['count()', '', 'count'],
+                ['min', 'timestamp', 'first_seen'],
+                ['max', 'timestamp', 'last_seen'],
+                ['uniq', key, 'values_seen'],
+            ],
+            orderby='-count',
+            referrer='tagstore.__get_tag_keys',
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_events.py]: Is this still accurate?

File: src/sentry/api/endpoints/organization_events.py [hunk 4]
```diff
@@ -170,6 +171,8 @@ def get(self, request, organization):
 
 
 class OrganizationEventsHeatmapEndpoint(OrganizationEventsEndpointBase):
+    NON_TAG_KEYS = frozenset(['project'])
```
Comment (lynnagara) [inline src/sentry/api/endpoints/organization_events.py]: Can we support this list of properties in an event? These are the fields we support elsewhere in Sentry. https://github.com/getsentry/sentry/blob/master/src/sentry/utils/snuba.py#L46-L115

Comment (lauryndbrown) [reply src/sentry/api/endpoints/organization_events.py]: Gotcha. I'll write up a ticket for this work, and do it in a different pr.

File: src/sentry/api/endpoints/organization_events.py (hunk 2 above)
Comment (lauryndbrown) [reply src/sentry/api/endpoints/organization_events.py]: I think in python it makes it harder to call from outside the class. I used to use them for "private" methods. I can use use one though. That looks like what we do.

File: src/sentry/api/endpoints/organization_events.py (hunk 1 above)
Comment (lauryndbrown) [reply src/sentry/api/endpoints/organization_events.py]: oh lol. thanks. :)

File: src/sentry/api/endpoints/organization_events.py (hunk 3 above)
Comment (lauryndbrown) [reply src/sentry/api/endpoints/organization_events.py]: nope good catch

**Review by markstory — APPROVED** [review]

**Pull request merged**